### ISSUE TITLE
Inconsistent formatting of copy/pasted mentions

### ISSUE THREAD

**Issue by micieslak** [status-im/status-desktop#7091]
# Bug Report

## Description

Mentions change formatting when copied and pasted.

Depends on [#7076](https://github.com/status-im/status-desktop/issues/7076) (but still can be reproduced without having #7076 fixed).

## Steps to reproduce

1. Write a mention in the chat input field
2. Add new line (shift + enter)
3. Select all (ctrl + A)
4. Copy (ctrl + C)
5. Unselect, position cursor at the end of the input
6. Paste (ctrl + V)

#### Expected behavior
Pasted mention(s) should be formatted as the original one(s).

#### Actual behavior
Pasted mention(s) has/have different formatting.

### Additional Information

https://user-images.githubusercontent.com/20650004/185593512-269dcadc-ee79-4c48-bfef-f80250c04b81.mp4

- Status desktop version:
- Operating System: Ubuntu

Implement as per Figma (pending confirmation by design): https://www.figma.com/file/Mr3rqxxgKJ2zMQ06UAKiWL/%F0%9F%92%AC-Chat%E2%8E%9CDesktop?node-id=20828%3A507322

**Comment by faust4exe** [comment]
This issue is related to rendering of text between underlines, which is a markdown notation, _ like this one_
Probably there is no easy way to ignore just this type of rendering while leaving all other.

![image](https://user-images.githubusercontent.com/6445843/194101825-498f4141-24bd-44c8-9df6-983429167bb1.png)

**Comment by benjthayer** [comment]
Chatting through formatting more broadly with @alexandraB99, for this issue I don't think there is much design can input on. Obviously from a design and UX perspective it would be ideal if there is no unintended variance in formatting when copy and pasting a mention.

More broadly (and this may warrant the creation of another issue) we may need to design for how formatting impacts @mentions. As a baseline for MVP we could emulate Discord:

<img width="727" alt="Screenshot 2022-12-12 at 18 37 08" src="https://user-images.githubusercontent.com/110033914/207126979-65923d31-bee6-4747-b995-7cc3b55cd5f0.png">

One thing to note in Discord is that the original formatting from the message is lost when you copy a sent message, paste it into the input and send it again. We could look to enhance this and retain the formatting when copied, pasted and resent but that feels like an enhancement and there are possibly more critical items to focus on.

<img width="435" alt="Screenshot 2022-12-12 at 18 39 00" src="https://user-images.githubusercontent.com/110033914/207127587-a843450f-4a8f-48aa-b3bf-ce7ad311037a.png">

cc @John-44 

I'd be happy to provide mockups of the Discord mention formatting if necessary - just let me know.

**Comment by alexandraB99** [comment]
@benjthayer should the same behavior apply also in the text input before sending the message?

**Comment by alexandraB99** [comment]
Design spec provided and will be implemented here https://github.com/status-im/status-desktop/issues/8464

**Comment by benjthayer** [comment]
Hi @alexandraB99 all of the formatting relating behaviours (including how they apply to mentions) are covered here:
https://www.figma.com/file/Mr3rqxxgKJ2zMQ06UAKiWL/%F0%9F%92%AC-Chat%E2%8E%9CDesktop?node-id=21104%3A500080&t=mwtupLfn99uaQFQJ-4

If I understand your question correctly, yes - when formatting text is copied from a sent message and pasted into the input, the formatting should be retained and shown alongside the relevant markdown in the input. That is shown here:

https://www.figma.com/file/Mr3rqxxgKJ2zMQ06UAKiWL/%F0%9F%92%AC-Chat%E2%8E%9CDesktop?node-id=20828%3A508271&t=mwtupLfn99uaQFQJ-4

Hope that answers your question - just shout if not.
